src/libraries/System.Private.CoreLib/src/System/Environment.cs

```diff
@@ -18,6 +18,11 @@ public static partial class Environment
         /// </summary>
         internal static bool IsSingleProcessor => ProcessorCount == 1;
 
+        /// <summary>
+        /// Gets whether the current process is authorized to perform security-relevant functions.
+        /// </summary>
+        public static bool IsPrivilegedProcess { get; } = IsAdminProcess();
```

Comment:
lazily initialize similar to https://github.com/dotnet/runtime/blob/5d5ecb5f34053befbe7e6f756325af07da6189c6/src/libraries/System.Private.CoreLib/src/System/Environment.cs#L114